easybuild/framework/easyconfig/types.py

```diff
@@ -124,20 +127,59 @@ def to_toolchain(tcspec):
 
     if isinstance(tcspec, list):
         if len(tcspec) == 2:
-            res = {'name': tcspec[0].strip(), 'version': tcspec[1].strip()}
+            if convert_to == TYPE_TUPLE:
+                res = (tcspec[0].strip(), tcspec[1].strip())
+            else:
+                res = {'name': tcspec[0].strip(), 'version': tcspec[1].strip()}
         else:
-            raise EasyBuildError("Can not convert list %s to toolchain dict. Expected 2 elements", tcspec)
+            errstr = "Can not convert list %s to name and version %s. Expected 2 elements"
+            raise EasyBuildError(errstr, tcspec, convert_to)
     else:
-        raise EasyBuildError("Conversion of %s (type %s) to toolchain dict is not supported", tcspec, type(tcspec))
+        errstr = "Conversion of %s (type %s) to name and version %s is not supported"
+        raise EasyBuildError(errstr, tcspec, type(tcspec), convert_to)
 
     return res
 
 
-# this uses to_toolchain, so it needs to be at the bottom of the module
+def to_dependency(dep):
+    """
+    Convert a dependency dict from the new YAML format {'name_version': 'lib, 1.2.8', 'toolchain': 'gcc, 4.8.2'}
+    to a tuple in the dependency format
+    """
+    depspec = []
+    if isinstance(dep, dict):
```

Comment:
the `_parse_dependency` method that will eventually parse all the values in `dependencies` can handle both a tuple value as the one you're creating here, or a `dict` value that contains `name`, `version`, `versionsuffix` and `toolchain` keys, so you should be able to simplify this significantly by just taking the dict and making sure `name` and `version` are what they should be; there's no need to mimic the tuple value we're usually using in `.eb` files